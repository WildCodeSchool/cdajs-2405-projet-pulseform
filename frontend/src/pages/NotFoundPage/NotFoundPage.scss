@import "@assets/variables";

.not-found-page {
  &__top {
    background-color: $tertiary;
    height: 40px;

    @media screen and (min-width: 768px) {
      height: 60px;
    }

    &__logo {
      position: relative;
      margin-left: 40px;
      border-radius: 50%;
      top: -2px;
      background-color: $tertiary;
      padding: 12px 8px;

      @media screen and (min-width: 768px) {
        display: none;
      }
    }

    &__logo-d {
      display: none;

      @media screen and (min-width: 768px) {
        display: block;
        position: relative;
        margin-left: 41px;
        border-radius: 50%;
        top: -21px;
        background-color: #4477e8;
        padding: 20px 13px;
      }
    }
  }

  &__main {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 90vh;

    @media screen and (min-width: 768px) {
      flex-direction: row;
    }

    &__img {
      width: 250px;
      height: 200px;
      margin-bottom: 20px;

      @media screen and (min-width: 768px) {
        width: 35%;
        height: 50%;
      }
    }

    &__content {
      @include h2-phone;
      text-align: center;
      padding: 0 30px;

      @media screen and (min-width: 768px) {
        @include logo-big-label;
        padding: 0;
      }

      &__button {
        margin-top: 20px;

        @media screen and (min-width: 768px) {
          margin-top: 40px;
        }
      }
    }
  }
}
