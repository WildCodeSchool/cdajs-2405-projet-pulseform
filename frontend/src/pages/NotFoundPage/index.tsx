import { useTranslation } from "react-i18next";

import ErrorBodyPage from "@components/organismes/ErrorBodyPage";

import blob404 from "@assets/images/blob-girl-404.svg";

function NotFoundPage() {
  const { t } = useTranslation();

  return (
    <ErrorBodyPage
      picture={blob404}
      altPicture={"blob404"}
      textError={t("PAGE_NOT_FOUND")}
    />
  );
}

export default NotFoundPage;
