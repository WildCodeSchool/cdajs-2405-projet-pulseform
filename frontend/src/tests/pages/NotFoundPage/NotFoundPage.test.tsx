import NotFoundPage from "@pages/NotFoundPage";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("NotFoundPage", () => {
  it("renders the NotFoundPage", () => {
    const tree = render(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
