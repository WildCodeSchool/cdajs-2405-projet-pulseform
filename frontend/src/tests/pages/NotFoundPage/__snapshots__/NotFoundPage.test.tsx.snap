// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > renders the NotFoundPage 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="error-body-page"
      >
        <div
          class="error-body-page__main"
        >
          <img
            alt="blob404"
            class="error-body-page__main__img"
            src="/src/assets/images/blob-girl-404.svg"
          />
          <div
            class="error-body-page__main__content"
          >
            <div>
              <p>
                OOPS
              </p>
              <p>
                PAGE_NOT_FOUND
              </p>
              <p>
                TRY_TO_GO_BACK
              </p>
            </div>
            <button
              class="basic-button error-body-page__main__content__button basic-button--orange"
              type="button"
            >
              BACK_TO_HOME
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="error-body-page"
    >
      <div
        class="error-body-page__main"
      >
        <img
          alt="blob404"
          class="error-body-page__main__img"
          src="/src/assets/images/blob-girl-404.svg"
        />
        <div
          class="error-body-page__main__content"
        >
          <div>
            <p>
              OOPS
            </p>
            <p>
              PAGE_NOT_FOUND
            </p>
            <p>
              TRY_TO_GO_BACK
            </p>
          </div>
          <button
            class="basic-button error-body-page__main__content__button basic-button--orange"
            type="button"
          >
            BACK_TO_HOME
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
